Lesson 11/HW11.cs

```diff
@@ -0,0 +1,30 @@
+namespace HW456.Lesson_11;
+
+/// <summary>
+/// Написать метод-расширение для массива целых чисел, который принимает как параметр строку-разделитель,
+/// а возвращает строку, в которая содержатся все элементы массива перечисленные через указанный разделитель.
+/// </summary>
+
+public static class HW11
+{
+    public static void Task()
+    {
+        Console.WriteLine("Input splitter: ");
+        string splitter = Console.ReadLine();
+
+        int[] array = Lesson11.GenerateArray();
+        
+        Console.WriteLine(splitter.ArrayWithSplitter(array));
+    }
+    public static string ArrayWithSplitter(this string splitter, int[] arr)
+    {
+        var numbersWithSplitter = String.Empty;
+
+        foreach (var i in arr)
+        {
+            numbersWithSplitter = numbersWithSplitter + i.ToString() + splitter;

```

Comment:
```suggestion
            numbersWithSplitter += i.ToString() + splitter;

```
А вообще такое не совсем оптимально с точки зрения реализации типа `string` (на каждое такое "расширение" строки создается новое значение в памяти). На почитать: [статья на хабре](https://habr.com/ru/articles/172627/).

Обычно такие проблемы решаются через `StringBuilder` (либо что то более низкоуровневое).

Тут на твое усмотрение, можно такое и не исправлять, учитывая что здесь точно проблем с памятью не будет, но для кругозора это полезно